### ISSUE TITLE
Manually add another provider

### ISSUE THREAD

**Issue by Timoms** [mageplaza/magento-2-social-login#139]
First of all: 
Thanks for the nice extension!

Im currently building my own software-ecosystem including an OAuth-Server.
Is there a quick way to add a new provider (say: a copy of the GitHub provider)? 

I've found some classes that could be modified but im not too sure if its working because nothing got displayed on the backend side.

I would greatly appreciate some tips!
Thanks in advance

**Comment by MpEdward** [comment]
I'm Edward from Mageplaza.
We using the already library for this module.
You can refer it in this link: https://github.com/hybridauth/hybridauth/tree/v2/hybridauth/Hybrid
If there any question please let me know.
Thank you!

**Comment by Timoms** [comment]
Thanks, Edward!
It worked perfectly. Good tip!

**Another question:** 
Do you think its possible to implement the auth-flow on the backend site?
I want to be able to login in the admin panel with my OAuth Identity Server.

Which classes do you think I need - or is there a good documentation with all the overrideable classes? 


Thanks for your help - im trying to contribute to your language pack as far as I can in order to give something back to you 👍

**Comment by MpEdward** [comment]
Hi @Timoms

Implementing the social network auth-flow in the backend site could be considerably challenging.
We are currently having this task under researching because unlike the frontend, Magento backend site has many security features and utilizing this type of authentication poses a number of threats as well as security risks to the backend site.

One of which could be mentioned is that the URL to the Magento backend site is being public when creating a social authentication app. This will leave your admin URL being accessible to the networks which could be a big digital hazard.
Others security threats are also very significant but I am not in the development team so I could not list all of them.

In short, implementing this feature could be feasible yet extremely difficult to complete and I myself do not have much insight toward this matter.
Really hope you will understand the situation.

**Comment by Desmond-Mageplaza** [comment]
Hi @Timoms,
It's been a while since your last response so I will close this issue.
Thank you for your interest in our extension
